src/app/screens/BookDetail/index.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import dataBooks from '../../../data/books.json';
+
+import BookDet from './layout';
+
+import './styles.css';
+
+export class BookDetail extends React.Component {
+  state = { book: '' };
+
+  componentWillMount() {
+    this.setState({ book: this.findBook(this.props.match.params.id) });
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.match.params.id !== this.props.match.params.id) {
+      this.setState({ book: this.findBook(nextProps.match.params.id) });
+    }
+  }
```

Comment:
`componentWillMount` and `componentWillReceiveProps` will be deprecated. 

Read this https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html and try to replace those functions with the new ones!